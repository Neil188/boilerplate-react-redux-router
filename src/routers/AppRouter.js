import React from 'react';
import { Router, Route, Switch } from 'react-router-dom';
import createHistory from 'history/createBrowserHistory';
import DashboardPage from '../components/DashboardPage';
import NotFoundPage from '../components/NotFoundPage';
import ConLoginPage from '../components/LoginPage';
import ConPrivateRoute from './PrivateRoute';
import ConPublicRoute from './PublicRoute';

export const history = createHistory();

const AppRouter = () => (
    <Router history={history} >
        <div>
            <Switch>
                <ConPublicRoute
                    exact
                    path='/'
                    component={ConLoginPage}
                />
                <ConPrivateRoute
                    path='/dashboard'
                    component={DashboardPage}
                />
                <Route
                    component={NotFoundPage}
                />
            </Switch>
        </div>
    </Router>
)

export default AppRouter;
