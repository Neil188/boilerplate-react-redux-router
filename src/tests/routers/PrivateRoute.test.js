import React from 'react';
import { shallow } from 'enzyme';
import { PrivateRoute, mapStateToProps } from '../../routers/PrivateRoute';

test('should render PrivateRoute for unauthorised user', () => {
    const props = {
        component: () => {},
        isAuthenticated: false
    }
    const wrapper = shallow(<PrivateRoute {...props} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.prop('component')().type.name).toBe('Redirect');
});

test('should render PrivateRoute for authorised user', () => {
    const props = {
        component: () => {},
        isAuthenticated: true
    }
    const wrapper = shallow(<PrivateRoute {...props} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.prop('component')().type).toBe('div');

});

test('State mapped to props', () => {
    const state = {
        auth: {
            uid: 1,
        }
    };
    const expected = {
        isAuthenticated: true
    }
    const props = mapStateToProps(state);
    expect(props).toEqual(expected);
});
