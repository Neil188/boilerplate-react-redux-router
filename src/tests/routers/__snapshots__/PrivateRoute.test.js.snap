// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PrivateRoute for authorised user 1`] = `
<Route
  component={[Function]}
/>
`;

exports[`should render PrivateRoute for unauthorised user 1`] = `
<Route
  component={[Function]}
/>
`;
